contracts/Rollup.sol

```diff
@@ -403,6 +457,86 @@ contract Rollup is RollupHelpers {
         }
     }
 
+    function disputeMMBatch(
+        uint256 _batch_id,
+        Types.MMCommitmentInclusionProof memory commitmentMP,
+        bytes memory txs,
+        Types.AccountMerkleProof[] memory accountProofs
+    ) public {
+        {
+            // check if batch is disputable
+            require(
+                !batches[_batch_id].withdrawn,
+                "No point dispute a withdrawn batch"
+            );
+            require(
+                block.number < batches[_batch_id].finalisesOn,
+                "Batch already finalised"
+            );
+
+            require(
+                (_batch_id < invalidBatchMarker || invalidBatchMarker == 0),
+                "Already successfully disputed. Roll back in process"
+            );
+        }
+
+        // verify is the commitment exits in the batch
+        {
+            require(
+                merkleUtils.verifyLeaf(
+                    batches[_batch_id].commitmentRoot,
+                    RollupUtils.MMCommitmentToHash(
+                        commitmentMP.commitment.stateRoot,
+                        commitmentMP.commitment.accountRoot,
+                        commitmentMP.commitment.txHashCommitment,
+                        commitmentMP.commitment.massMigrationMetaInfo.tokenID,
+                        commitmentMP.commitment.massMigrationMetaInfo.amount,
+                        commitmentMP
+                            .commitment
+                            .massMigrationMetaInfo
+                            .withdrawRoot,
+                        commitmentMP
+                            .commitment
+                            .massMigrationMetaInfo
+                            .targetSpokeID,
+                        commitmentMP.commitment.signature
+                    ),
+                    commitmentMP.pathToCommitment,
+                    commitmentMP.siblings
+                ),
+                "Commitment not present in batch"
+            );
+
+            require(
+                commitmentMP.commitment.txHashCommitment != ZERO_BYTES32,
+                "Cannot dispute blocks with no transaction"
+            );
+        }
+
+        bytes32 updatedBalanceRoot;
+        bool isDisputeValid;
+        bytes32 txRoot;
+        (updatedBalanceRoot, txRoot, isDisputeValid) = rollupReddit
+            .processMMBatch(commitmentMP.commitment, txs, accountProofs);
```

Comment:
[later] I think this has the same problem as #177. The state root in the commitment is the "post" state root, while the processBatch needs "pre" state root.